### ISSUE TITLE
Can it work with AsciiMath?

### ISSUE THREAD

**Issue by iacchus** [the-turk/flarum-mathren#35]
Sources:

https://docs.mathjax.org/en/latest/input/asciimath.html

http://asciimath.org/

**Comment by hasan-ozbey** [comment]
No, KaTeX does not support AsciiMath input but we can use [this library](https://github.com/christianp/asciimath2tex) which converts AsciiMath to TeX.

**Comment by hasan-ozbey** [comment]
Done! You can now allow this syntax from the extension's settings. 

Don't forget to clear your cache (`php flarum cache:clear`) after changing its state.

**Comment by iacchus** [comment]
Awesome! :) Thank you very much.

**Comment by iacchus** [comment]
Bug reports for it are ok?

I've seen that AsciiMath and KaTeX don't work together, or one works, or the other one.

**Comment by hasan-ozbey** [comment]
> Bug reports for it are ok?

of course!

> I've seen that AsciiMath and KaTeX don't work together, or one works, or the other one.

can you provide a formula for me to take a look at it?

____

Oh I see what you mean now, you can't do proper TeX when you have AsciiMath enabled - but this is expected. Perhaps we need some custom delimiters to parse AsciiMath-only? I see the official delimiter for this syntax is backtick but we can't use that for obvious reasons. 

I think I'll add a new setting for AsciiMath delimiters so you can set this to anything you want and I'll add a new checkbox right under the text editor buttons - labaled as "AsciiMath only". So when you click it, it'll switch the TeX delimiters into AsciiMath delimiters.

**TLDR;** there will be standalone delimiters for AsciiMath, and a new checkbox under the text editor buttons labeled as "AsciiMode-only".

**Comment by hasan-ozbey** [comment]
check out `1.0.7` - and again, don't forget to clear your cache 😉

See it in action here: https://i.imgur.com/PEG1psE.mp4

...equations that I've used in that video:

AsciiMath:
`sum_(i=1)^n i^3=((n(n+1))/2)^2`

LaTeX:
`\sum_{i=1}^{n} i^3=\left(\frac{n(n+1)}{2}\right)^2`

**Comment by iacchus** [comment]
Awesome! it is very cool
